reactioncommerce/reaction
# refactor: refunds section of Orders 2.0

Addresses the refunds portion of #5157 
Impact: **minor**  
Type: **refactor**

## Issue
This PR is part of refactoring the Orders Operator UI to deal with multiple fulfillment groups, graphql, etc, to make it 2.0 ready. The majority of the Orders 2.0 UI has already been [tested and merged into Reaction](https://github.com/reactioncommerce/reaction/pull/5158).

This PR splits off the refunds portion of the overall work to make it easier to test in smaller chunks. This PR only addresses at the UI portion of refunds - the entirety of the refunds tab. The GraphQL queries and mutations related to refunds have already been tested and merged (#5352, #5354), and don't need to be addressed in this PR.

## Breaking changes
None

## Testing
- Place two orders (six total) of each of the following: 
- - Using only the Stripe payment method
- - Using only the example payment method
- - Using multiple payments, a combination of Stripe and Example (two or more payments)

- Navigate to the Orders Operator UI
- Click on an order, and then once on the single order page, change the URL from `/orders/` to `/orders-beta/` to see the new UI.

- Once an order is open, `Capture` all payments on the order.

- Then, navigate to the refunds tab. It should look something like this:

![Order_Details_for_order_reference__Y3imj7gnML4JtX29B](https://user-images.githubusercontent.com/4482263/62333936-709f2b00-b47a-11e9-8be4-9afdf8913a69.png)

At this point, do the following tests:
1. Refund the maximum amount of the single stripe payment
1. Refund the maximum amount of the single example payment method
1. refund the maximum amount of all methods in the multiple payment order (you can 

- After all those are done, confirm the input box disappears, and you can no longer create a refund.
- Confirm you can see previous refunds in the appropriate section

Then:
1. Refund a partial amount of a single stripe payment
1. Refund the partial amount of the single example payment method
1. refund the partial amount of all methods in the multiple payment order (you can 

- At this point, see that there is a line of text that tells you how much has been refunded, and how much is available to refund on each payment

Then:
1. attempt to refund more than the remaining amount on all payments
1. The UI will "let" you, but the server side will only allow the remaining amount and complete the refund

#### Adding translations for Chips:
Add orderStatusLabels to the Shops collection shop object. This will provide translations for all the Chip style order status badges in the Orders Fulfillment UI.
```
"orderStatusLabels": {
    "new" : [ 
        {
            "language" : "en",
            "label" : "Order received"
        }, 
        {
            "language" : "es",
            "label" : "Recibido"
        }
    ],
    "coreOrderWorkflow/processing" : [ 
        {
            "language" : "en",
            "label" : "Order received"
        }, 
        {
            "language" : "es",
            "label" : "Recibido"
        }
    ],
    "coreOrderWorkflow/completed" : [ 
        {
            "language" : "en",
            "label" : "Shipped"
        }, 
        {
            "language" : "es",
            "label" : "Enviado"
        }
    ],
    "coreOrderWorkflow/canceled" : [ 
        {
            "language" : "en",
            "label" : "Canceled"
        }, 
        {
            "language" : "es",
            "label" : "Cancelado"
        }
    ],
    "coreOrderWorkflow/picked" : [ 
        {
            "language" : "en",
            "label" : "Picked"
        }
    ],
    "coreOrderWorkflow/packed" : [ 
        {
            "language" : "en",
            "label" : "packed"
        }
    ],
    "coreOrderWorkflow/shipped" : [ 
        {
            "language" : "en",
            "label" : "shipped"
        }
    ]
}
```
